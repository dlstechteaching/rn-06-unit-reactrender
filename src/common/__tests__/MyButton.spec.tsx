import React from "react";
import { TouchableOpacity } from "react-native";
import renderer, { ReactTestRenderer } from "react-test-renderer";
import { MyButton } from "../MyButton";

describe("MyButton tests", () => {
  let component: ReactTestRenderer;

  const onPressMock = jest.fn();
  it("should render properly", () => {
    givenComponent();
    thenItRendersProperly();
  });

  it("should call onPressMock", () => {
    givenComponent();
    whenButtonIsPressed();
    thenOnPressHasBeenCalled();
  });

  function givenComponent() {
    component = renderer.create(
      <MyButton label={"label"} onPress={onPressMock} />
    );
  }

  function whenButtonIsPressed() {
    component.root.findByType(TouchableOpacity).props.onPress();
  }

  function thenItRendersProperly() {
    expect(component).toMatchSnapshot();
  }

  function thenOnPressHasBeenCalled() {
    expect(onPressMock).toHaveBeenCalled();
  }
});
