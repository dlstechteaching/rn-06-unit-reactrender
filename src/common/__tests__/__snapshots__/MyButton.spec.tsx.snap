// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyButton tests should render properly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(92, 45, 145, 1)",
      "borderBottomEndRadius": 15,
      "borderRadius": 3,
      "marginVertical": 20,
      "opacity": 1,
      "padding": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "textAlign": "center",
      }
    }
  >
    label
  </Text>
</View>
`;
